src/site/topics/bags/bag-implementations.rst

```diff
@@ -2,419 +2,299 @@
 Bag Implementations
 *******************
 
-* :doc:`We already know what we want to do with our bags <bag-adt>`
-* We also extended the ``Bag`` to make more specialized bags
+* Given the definition of a ``Bag``
+* And given the definitions of the specialized bags
+
     * ``IndexedBag``
     * ``SortedBag``
```

Comment:
A `MapBag` may actually be a half decent assignment option. That is, a bag backed by a `Map<T, Integer>`.